scopes/workspace/workspace/workspace.ts

```diff
@@ -1277,6 +1280,73 @@ export class Workspace implements ComponentFactory {
       await this.importObjects();
       this.logger.consoleSuccess();
     }
+    return this.installModules(options);
+  }
+
+  private async getComponentsWithDependencyPolicies() {
+    const allComponentIds = await this.listIds();
+    const componentConfigFiles: Record<string, ComponentConfigFile> = {};
+    const componentPoliciesById: Record<string, any> = {};
+    (
+      await Promise.all<ComponentConfigFile | undefined>(
+        allComponentIds.map((componentId) => this.componentConfigFile(componentId))
+      )
+    ).forEach((component) => {
+      if (component == null) return;
+      const depResolverAspect = component?.aspects.get(DependencyResolverAspect.id);
+      if (depResolverAspect == null || component == null) return;
+      const componentId = component.componentId.toString();
+      componentConfigFiles[componentId] = component;
+      // eslint-disable-next-line
+      componentPoliciesById[componentId] = depResolverAspect.legacy.rawConfig['policy'];
+    });
+    return {
+      componentConfigFiles,
+      componentPoliciesById,
+    };
+  }
+
+  async updateDependencies() {
+    const { componentConfigFiles, componentPoliciesById } = await this.getComponentsWithDependencyPolicies();
+    const variantPatterns = this.variants.raw();
+    const outdatedPkgs = await this.dependencyResolver.getOutdatedPkgs({
+      rootDir: this.path,
+      variantPatterns,
+      componentPoliciesById,
+    });
+    const { updatedVariants, updatedComponents } = this.dependencyResolver.applyUpdates(outdatedPkgs, {
+      variantPatterns,
+      componentPoliciesById,
+    });
+    await this.updateVariants(variantPatterns, updatedVariants);
+    const updatedComponentConfigFiles = Object.values(pick(componentConfigFiles, updatedComponents));
+    await this.saveManyConfigFiles(updatedComponentConfigFiles);
+    await this._reloadConsumer();
+    return this.installModules({ dedupe: true });
+  }
+
+  private updateVariants(variantPatterns: Record<string, any>, updateVariantPolicies: string[]) {
```

Comment:
rename to something that show that it handles not any updateVariants, but a very specific type of update - update variants policies